delete extra drpdown & alter drpdown js

bethanyhaubert opened the pull request.

bethanyhaubert commented:
tyring to get onchange to work. looked at a few help pages. http://help.dottoro.com/ljoiurdq.php

bethanyhaubert commented:
tried both of these:
https://developer.mozilla.org/en-US/docs/Web/Events/change
http://help.dottoro.com/ljoiurdq.php

sumeetjain commented on app/models/client.rb:
```diff
@@ -8,6 +8,7 @@ class Client < ActiveRecord::Base
    def recent_touches
     self.touches # TODO Make this only return one touch for the client.
    end
+   
```
Remove trailing whitespace

sumeetjain replied:
Trailing whitespace detected.

sumeetjain commented on app/controllers/pages_controller.rb:
```diff
@@ -1,18 +1,19 @@
     end
   end
-
+  
```
Remove trailing whitespace

sumeetjain replied:
Trailing whitespace detected.

sumeetjain commented on app/controllers/pages_controller.rb:
```diff
@@ -1,18 +1,19 @@
 class PagesController < ApplicationController
+
```
Extra empty line detected at class body beginning.

sumeetjain commented on app/controllers/pages_controller.rb:
```diff
@@ -1,18 +1,19 @@
-        @touches = nil
+      @clients = []
+      @business.clients.each do |client|
+        if Touch.where({"client_id" => @current_client}) != nil
```
Prefer `!expression.nil?` over `expression != nil`.

sumeetjain commented on app/assets/javascripts/display_selected_client_from_dropdown.js:
```diff
@@ -3,24 +3,30 @@ jQuery(document).ready(function(){
-      var selectedId = document.getElementById("dropdown_select");
-      var presentation_area = document.getElementById("current_client");
+    // if (event.target.matches(".dropdown_submit")){
+    if (event.target.matches("dropdown_select")){
```
This needs to be `event.target.matches("#dropdown_select")` if it's finding something with an ID of `dropdown_select`. Doing `matches("dropdown_select")` means you're looking for an element with a **tag** called `<dropdown_select>`.

bethanyhaubert commented:
ok, added the '#'.  still getting error. https://www.dropbox.com/s/olrt12fzs57i1wo/Screenshot%202016-03-31%2014.04.47.png?dl=0

bethanyhaubert commented:
i don't get how
 ```
function wordsWordsWords (more_words) {
}
```
should be used

sumeetjain commented:
Is some of this copied from a tutorial or something? I think it might not be 100% compatible with your goal. Can you try removing the `function OnSelectionChange (select)` scope entirely?

And then I think `var selectedOption = select.options[select.selectedIndex];` would be `var selectedOption = event.target.options[select.selectedIndex];`

bethanyhaubert commented:
still not working

sumeetjain commented:
`if (event.target.matches(".dropdown"))` is looking for elements that have a **class** of "dropdown". Is that your intention? I thought you were trying to check for clicks occurring on `<select id="dropdown_select"`...>`. That's an **ID**, so you'd need `if (event.target.matches("#dropdown"))`.

Secondly, delete ` onchange="OnSelectionChange (this)"` from the `<select>` tag. It's an outdated technique that's been replaced by `addEventListener`.

bethanyhaubert commented:
can you just tell me what's wrong? i've changed something each time you've given feedback and i've changed it so much there's no way it's anywhere near what it was originally and i'm over it.
